### PR TITLE
Remove redundant upper case aliases for `median`, `first_value` and `last_value`

### LINKED ISSUE #10695: Use lowercase `name` for aggregate function
### Is your feature request related to a problem or challenge?

`first_value` and `last_value` window functions currently report their names as `FIRST_VALUE` but also have a lower case alias:

https://github.com/apache/datafusion/blob/98cd19ef1cf52aef647e3e43fcc132a93028d1a2/datafusion/functions-aggregate/src/first_last.rs#L75

part of #8708

### Describe the solution you'd like

I think we would like to avoid the aliases and simply have a single name, similarly how the other functions work

TODO
- Count

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by goldmedal**
## Which issue does this PR close?

Closes #10695 .

## What changes are included in this PR?
Fix `median`, `first_value` and `last_value`

## Are these changes tested?
Yes

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/functions-aggregate/src/first_last.rs [hunk 1]
```diff
@@ -72,7 +72,7 @@ impl Default for FirstValue {
 impl FirstValue {
     pub fn new() -> Self {
         Self {
-            aliases: vec![String::from("first_value")],
+            aliases: vec![],
```
Comment (alamb) [inline datafusion/functions-aggregate/src/first_last.rs:75]: A minor comment is we can probably remove the `aliases` field entirely and just return an empty slice

```rust
    fn aliases(&self) -> &[String] {
        &[]
    }
```

File: datafusion/functions-aggregate/src/lib.rs [hunk 2]
```diff
@@ -109,13 +109,13 @@ mod tests {
         let mut names = HashSet::new();
         for func in all_default_aggregate_functions() {
             assert!(
-                names.insert(func.name().to_string()),
+                names.insert(func.name().to_string().to_lowercase()),
```
Comment (jayzhan211) [inline datafusion/functions-aggregate/src/lib.rs:112]: Can we not convert to lowercase here? If someone changes the name to uppercase again, the function in the function map is not the same as the function name, and it is easy to introduce a bug.

Comment (goldmedal) [reply datafusion/functions-aggregate/src/lib.rs:112]: Thanks @jayzhan211 
I think `to_lowercase` is used to ensure that the function name and its alias won't be duplicates when case-insensitive (e.g., unquoted identifier). If someone changes the function name to uppercase, the test will pass. However, they may find that they can't query this function in lowercase. They might try to add a lowercase alias or change the function name back to lowercase. I think this is the purpose of this test.

Comment (jayzhan211) [reply datafusion/functions-aggregate/src/lib.rs:112]: Oh, it is the test, then it is fine

**Review by jayzhan211 — APPROVED** [review]
👍

**Comment by goldmedal** [comment]
Thanks again @alamb @jayzhan211

**Pull request merged**